test/api_get_robotoff_test.dart

```diff
@@ -58,6 +59,22 @@ void main() {
       }
     });
 
+    test('get questions with a given insight type', () async {
```

Comment:
Will there always be category questions for this specific product?

A stronger test would perhaps
* extract insights without barcodes (`getQuestions`) - we get barcodes and insight types
* check with `getProductQuestions` (with barcode and _insight_ parameter) that we do have a question for each barcode/insight type

Reply:
Indeed, that's a better implementation.
I will do it that way

Reply:
I have changed the test @monsieurtanuki, please tell me if that is what you were expected

Reply:
Thank you @g123k, that's more or less what I had in mind!